packages/core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -98,21 +108,26 @@ const DatePicker: VibeComponent<DatePickerProps, HTMLElement> = forwardRef<HTMLD
       [firstDayOfWeek]
     );
 
-    const changeCurrentDateFromMonthYearView = useCallback((date: Moment | null) => {
-      setOverrideDateForView(date);
-      setIsMonthYearSelection(false);
-    }, []);
+    const changeCurrentDateFromMonthYearView = useCallback(
+      (newDate: Moment | null) => {
+        const oldDate = overrideDateForView || date;
+        setOverrideDateForView(newDate);
+        setIsMonthYearSelection(false);
+        yearFunc(oldDate, moment(newDate).year());
+      },
+      [overrideDateForView, date, yearFunc]
+    );
 
     const renderMonthYearSelection = useCallback(() => {
       return (
         <YearPicker
           data-testid={dataTestId || getTestId(ComponentDefaultTestId.DATEPICKER_YEAR_SELECTION, id)}
-          selectedDate={date}
+          selectedDate={overrideDateForView || date}
```

Comment:
Why are we changing the selected date?

Reply:
In order to fix the bug in the years select that it wasn't highlighting the newly selected year.